hangman.py

```diff
@@ -0,0 +1,427 @@
+    for symbol in secret_word:
+        new_set.add(symbol)
+    if letters_guessed | new_set == letters_guessed:
+        return True
+    else:
+        return False
+
+
+
+def get_guessed_word(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string, comprised of letters, underscores (_), and spaces that represents
+      which letters in secret_word have been guessed so far.
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    new_list = []
+    for symbol in secret_word:
+        if symbol in letters_guessed:
+            new_list.append(symbol)
+        else:
+            new_list.append('_ ')
+    str = ''.join(new_list)
+    return str
+
+
+
+def get_available_letters(letters_guessed):
+    '''
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string (of letters), comprised of letters that represents which letters have not
+      yet been guessed.
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    al = str(string.ascii_lowercase)
+    new_list = []
+    for symbol in al:
+        new_list.append(symbol)
+
+    for symbol in letters_guessed:
+        new_list.remove(symbol)
+    str_not_used_letters = ''.join(new_list)
+    return str_not_used_letters
+
+
+def score(secret_word, letters_guessed):
+    letters_guessed = set(letters_guessed)
+    new_set = set()
+    for symbol in secret_word:
+        new_set.add(symbol)
+    length = len(new_set)
+    return length
+    
+    
+
+def hangman(secret_word):
+    '''
+    secret_word: string, the secret word to guess.
+    
+    Starts up an interactive game of Hangman.
+    
+    * At the start of the game, let the user know how many 
+      letters the secret_word contains and how many guesses s/he starts with.
+      
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+    
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+    
+    * The user should receive feedback immediately after each guess 
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the 
+      partially guessed word so far.
+    
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+
+    def letters_in_secret_word(secret_word):
+        letters_in_secret_word = []
+        for letters in secret_word:
+            if letters in letters_in_secret_word:
+                continue
+            letters_in_secret_word.append(letters)
+        return letters_in_secret_word
+
+    letters_in_secret_word = letters_in_secret_word(secret_word)
+    list_of_en_golos = ["a", "e", "i", "o", "u"]
```

Comment:
Do you think that list is the best type for it?

Reply:
done